static/js/compose_ui.js

```diff
@@ -153,6 +153,13 @@ export function make_compose_box_full_size() {
 
     $("#compose").addClass("compose-fullscreen");
 
+    // Set the `top` property of compose box.
+    const compose_top =
+        $("#navbar_alerts_wrapper").height() +
+        $(".header").height() +
+        Number.parseInt($(".header").css("paddingBottom"), 10);
+    $("#compose").css("top", compose_top + "px");
```

Comment:
This sort of javascript-based approach feels fragile.  Is there a way that we can refactor the HTML a bit to just make the 100% mean 100% of the space below the navbar panels?

Reply:
Actually, the compose has `position: fixed` property, so no matter what we do with the HTML, 100% for the compose will always be 100% of the entire viewport. This can be solved by moving the compose box from `.column-middle` to `.app-main` and using `position: absolute` there, but this seems to be a lot of work which may also lead to many unnecessary bugs.

The code I used above is actually how we set the top of `#floating_recipient_bar` [here](https://github.com/zulip/zulip/blob/1295b47e3b9042c4ea7cb10d9928499ba141d193/static/js/navbar_alerts.js#L27).

Reply:
I agree with @garg3133 , I also tried the same way. But the `#compose` has `position: fixed` which is causing the problem.

Reply:
Well, I can't say I have a clear sense as to whether we changing to `position: absolute` would be better; that may be a discussion worth taking to #frontend, but probably not a blocker for this.

We should record that background in the comments here, however.

Reply:
Updated. Please check the comment.